lib/main_app.rb

```diff
@@ -9,19 +9,23 @@ def run
     @game = GameLogic.new
     @console_output = ConsoleOutput.new
 
-    until @game.is_there_winner? do
+    until @game.board_full? || @game.is_there_winner? do
       puts(@console_output.welcome_message())
       puts(@console_output.show_board(@game))
 
       valid_options = @game.open_spaces.map(&:to_s)  
 
       user_choice = @questions.validate_user_input("PICK A SPACE #{@game.current_player()}", valid_options)
-      
-      @game.take_turn(user_choice) 
+      @game.take_turn(user_choice)
+    end
 
+    if @game.is_there_winner?
+      puts(@console_output.show_board(@game)) 
+      puts("GAME OVER: #{@game.is_there_winner?} WINS!!")
+    else
+      puts(@console_output.show_board(@game))
+      puts(@console_output.tie_message)
```

Comment:
```suggestion
    puts(@console_output.show_board(@game)) 
    if @game.is_there_winner?
      puts("GAME OVER: #{@game.is_there_winner?} WINS!!")
    else
      puts(@console_output.tie_message)
```
You output the gameboard in both the if and else block. Let's move that above to remove duplication.